src/crypto/cn/CryptoNight_x86.h

```diff
@@ -293,9 +293,10 @@ template<Algorithm::Id ALGO, bool SOFT_AES, int interleave>
 static NOINLINE void cn_explode_scratchpad(cryptonight_ctx *ctx)
 {
     constexpr CnAlgo<ALGO> props;
+    constexpr bool IS_CN_HEAVY = props.isHeavy();
 
 #   ifdef XMRIG_VAES
-    if (!SOFT_AES && !props.isHeavy() && cn_vaes_enabled) {
+    if (cn_vaes_enabled && !SOFT_AES && !IS_CN_HEAVY) {
```

Comment:
cn_vaes_enabled (non compile-time check) must be the last in list, or compiler won't be able to remove the whole `if` block when one of the other two compile-time checks returns false.